backward.hpp

```diff
@@ -1737,59 +1747,97 @@ class Printer {
 
 	template <typename ST>
 		FILE* print(ST& st, FILE* os = stderr) {
+			std::stringstream ss;
+			Colorize colorize(ss);
+			if (color) {
+				colorize.activate_if_tty(os);
+			}
+			print(st, ss, colorize);
+			fprintf(os, "%s", ss.str().c_str());
+			return os;
+		}
+
+	template <typename ST>
+		void print(ST& st, std::ostream& os) {
 			Colorize colorize(os);
 			if (color) {
-				colorize.init();
+				colorize.activate();
```

Comment:
we cannot activate color on ostream by default as there is no way to know if the output is a terminal or not. You must take a `bool color=false` as argument. `this->color` here is more a hint from the user enabling tty color detection.